src/components/NodeEditor/NodeEditor.tsx

```diff
@@ -436,51 +436,65 @@ export default class NodeEditor extends React.PureComponent<NodeEditorProps, Nod
      * Returns existing action (if any), or a bare-bones representation of the form's action.
      */
     private getAction(): AnyAction {
+        let uuid: string;
+
         if (this.props.action) {
             if (
                 this.props.action.type === this.state.config.type ||
                 (this.state.config.aliases &&
                     this.props.action.type === this.state.config.aliases[0])
             ) {
                 return this.props.action;
+            } else {
+                ({ uuid } = this.props.action);
             }
         }
 
-        let action: Action = {
+        let defaultAction: Action = {
```

Comment:
Ensuring we're using the existing action's `UUID` if applicable. Important for `FlowMutator`'s ' `updateAction` method